eth/protocols/eth/handlers.go

```diff
@@ -275,6 +276,13 @@ func handleNewBlock(backend Backend, msg Decoder, peer *Peer) error {
 		log.Warn("Propagated block has invalid external transaction", "have", hash, "exp", ann.Block.EtxHash())
 		return nil // TODO(karalabe): return error eventually, but wait a few releases
 	}
+	// Dom nodes need to validate the subordinate manifest against the subordinate's manifesthash
```

Comment:
Same comment as above. Can we reuse the body validator here too? It feels silly to rewrite the same code 3+ times

Reply:
same as above.

Reply:
we can maybe create a function just to validate the body piece and use in both places.

Reply:
Created an issue to come back to this later: https://github.com/dominant-strategies/go-quai/issues/571